js/main.js

```
@@ -0,0 +1,77 @@
+const Keys = {
+  ALT: `Alt`,
+  LEFT_ARROW: `ArrowLeft`,
+  RIGHT_ARROW: `ArrowRight`
+};
+
+const DEFAULT_SCREEN_INDEX = 0;
+const mainElement = document.querySelector(`main.central`);
+
+let screenIndex = DEFAULT_SCREEN_INDEX;
+let map = {};
+
+const screenTemplates = document.querySelectorAll(`template`);
+const contentOfScreenTemplates = Array.from(screenTemplates).map((template) => template.content);
+
+/**
+ * Change index of screen to th new one and render screen with new index
+ *
+ * @param {number} newScreenIndex
+ * **/
+const switchScreen = (newScreenIndex) => {
+  if ((newScreenIndex < 0) || (newScreenIndex >= contentOfScreenTemplates.length)) {
+    return;
+  }
+
+  screenIndex = newScreenIndex;
+  const screenElement = contentOfScreenTemplates[screenIndex].cloneNode(true);
+  while (mainElement.hasChildNodes()) {
+    mainElement.removeChild(mainElement.lastChild);
+  }
+  mainElement.appendChild(screenElement);
+};
+
+/**
+ * Keydown event handler
+ *
+ * @param {KeyboardEvent} evt
+ * **/
+const onSwitchScreenKeydown = (evt) => {
+  evt = evt || event; // to deal with IE
+  if (Object.values(Keys).indexOf(evt.key) === -1) {
```

Ненужное условие. Правильно проверять те ключи, которые нам нужны и выполнять действие.

При тустировании я обнаружил, что переключение не работает корректно. У меня снова переключается по клавишам Влево и Вправо без альт.

---

> Правильно проверять те ключи, которые нам нужны и выполнять действие.
Я попытался следовать принципу fail fast :)

---

>  У меня снова переключается по клавишам Влево и Вправо без альт.

Мне, по работе, часто приходится тестить чужой код. Обычно - получается много ошибок выявлять. Но в данном случае - я не могу это воспроизвести. :)

---

> Я попытался следовать принципу fail fast :)
Да. но ты здесь в принципе и задаёшь условие по которому и только по которому выполнять действия, так что остальные клавиши будут откинуты :)